wit/deps/spin-postgres@4.0.0/postgres.wit

```diff
@@ -0,0 +1,130 @@
+package spin:postgres@4.0.0;
+
+interface postgres {
+  /// Errors related to interacting with a database.
+  variant error {
+      connection-failed(string),
+      bad-parameter(string),
+      query-failed(string),
+      value-conversion-failed(string),
+      other(string)
+  }
+
+  /// Data types for a database column
+  enum db-data-type {
+      boolean,
+      int8,
+      int16,
+      int32,
+      int64,
+      floating32,
+      floating64,
+      str,
+      binary,
+      date,
+      time,
+      datetime,
+      timestamp,
+      uuid,
+      jsonb,
+      decimal,
+      range32,
+      range64,
+      array-int32,
+      array-int64,
+      array-str,
+      other,
+  }
+
+  /// Database values
+  variant db-value {
+      boolean(bool),
+      int8(s8),
+      int16(s16),
+      int32(s32),
+      int64(s64),
+      floating32(f32),
+      floating64(f64),
+      str(string),
+      binary(list<u8>),
+      date(tuple<s32, u8, u8>), // (year, month, day)
```

Comment:
A tuple that needs a comment to explain its elements sounds like it wants to be a record . 🙂